### PR TITLE
Client side replication

### PR DESCRIPTION
**Pull request by RicoGit**

### CODE REVIEW

File: client/src/main/scala/fluence/client/ClientReplicationWrapper.scala [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import monix.eval.Task
+
+import scala.language.higherKinds
+
+/**
+ * Wrapper with dataset replication. Allows writes data to all nodes and reads data if at least one node is alive.
+ * This is naive implementation of replication and it will be removed in future.
+ */
+class ClientReplicationWrapper[K, V](
+    datasetReplicas: List[(ClientDatasetStorageApi[Task, K, V], Contact)]
```
Comment (alari) [inline client/src/main/scala/fluence/client/ClientReplicationWrapper.scala]: Probably `Contact` should be at the left side, and it even might be a map.

File: client/src/main/scala/fluence/client/ClientReplicationWrapper.scala [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+   * @param value The value associated with the specified key
+   * @return returns old value if old value was overridden, None otherwise.
+   */
+  override def put(key: K, value: V): Task[Option[V]] = {
+    Task.gatherUnordered(
+      datasetReplicas
+        .map { case (store, _) ⇒ store.put(key, value) }
+    ).map { seq ⇒
+        logger.info(s"$key and $value was written to $replicationFactor nodes")
+        seq.head
```
Comment (alari) [inline client/src/main/scala/fluence/client/ClientReplicationWrapper.scala]: Is it safe?

Comment (RicoGit) [reply client/src/main/scala/fluence/client/ClientReplicationWrapper.scala]: yes

File: client/src/main/scala/fluence/client/ClientReplicationWrapper.scala (hunk 1 above)
Comment (RicoGit) [reply client/src/main/scala/fluence/client/ClientReplicationWrapper.scala]: Contract is for nice logging only, there is no needed Map

File: client/src/main/scala/fluence/client/ClientReplicationWrapper.scala (hunk 2 above)
Comment (alari) [reply client/src/main/scala/fluence/client/ClientReplicationWrapper.scala]: Could you please add comment about why is it safe

File: client/src/main/scala/fluence/client/ClientReplicationWrapper.scala (hunk 1 above)
Comment (alari) [reply client/src/main/scala/fluence/client/ClientReplicationWrapper.scala]: Then why not to document that?

File: client/src/main/resources/reference.conf [hunk 3]
```diff
@@ -1,8 +1,18 @@
...
+  kademlia {
+    maxBucketSize: 16
+    maxSiblingsSize: 32
+    parallelism: 3
+    pingExpiresIn: 1 hour
+  }
+
+  network {
+    join {
+      seeds: [
```
Comment (alari) [inline client/src/main/resources/reference.conf]: seeds are in `fluence.network.seeds` now

File: client/src/main/scala/fluence/client/FluenceClient.scala [hunk 4]
```diff
@@ -147,6 +155,58 @@ class FluenceClient(
...
+      key ← Key.fromKeyPair(ac.kp)
+      bcOp ← contracts.find(key).attempt.map(_.toOption)
+      dataStorages ← bcOp match {
+
+        case Some(basicContract) ⇒ //create storage from existed contract
+          for {
+            nodes ← findContactsOfAllParticipants(basicContract)
+            datasets ← Task.sequence(
+              nodes.map( contact ⇒
+                addNonEncryptedDataset(ac, contact, Some(ClientState(basicContract.executionState.merkleRoot.toArray)))
```
Comment (alari) [inline client/src/main/scala/fluence/client/FluenceClient.scala]: We should move to `ByteVector` for merkle roots everywhere

Comment (RicoGit) [reply client/src/main/scala/fluence/client/FluenceClient.scala]: Ok

File: client/src/main/resources/reference.conf (hunk 3 above)
Comment (RicoGit) [reply client/src/main/resources/reference.conf]: Without this client isn't starts. Are you sure?

File: client/src/main/scala/fluence/client/ClientReplicationWrapper.scala (hunk 1 above)
Comment (RicoGit) [reply client/src/main/scala/fluence/client/ClientReplicationWrapper.scala]: Cause it not necessary )

**Pull request merged**